test/CQRS/LeanCode.CQRS.MassTransitRelay.Tests/Middleware/EventsPublisherMiddlewareWithDefaultConfigurationTests.cs

```diff
@@ -0,0 +1,102 @@
+#nullable enable
+using System.Linq;
+using System.Security.Claims;
+using FluentAssertions;
+using LeanCode.IntegrationTestHelpers;
+using Xunit;
+
+namespace LeanCode.CQRS.MassTransitRelay.Tests.Middleware;
+
+public sealed class EventsPublisherMiddlewareWithDefaultConfigurationTests : EventsPublisherMiddlewareBaseTests
+{
+    public EventsPublisherMiddlewareWithDefaultConfigurationTests()
+        : base(cfg => { }) { }
+
+    [Fact]
+    public async Task Does_not_propagate_anything_when_the_user_is_unauthorized()
+    {
+        await Harness.Start();
+        await SendAsync(new TestCommand(), null);
+        await WaitForProcessing();
+        (await Harness.Published.SelectAsync<TestEvent1>().ToListAsync())
+            .Should()
+            .ContainSingle()
+            .Which.Context.Headers.Should()
+            .BeEmpty();
+    }
+
+    [Fact]
+    public async Task Does_not_propagate_anything_in_consumers_when_the_user_is_unauthorized()
+    {
+        await Harness.Start();
+        await SendAsync(new TestCommand(), null);
+        await WaitForProcessing();
+        (await Harness.Published.SelectAsync<TestEvent2>().ToListAsync())
+            .Should()
+            .ContainSingle()
+            .Which.Context.Headers.Should()
+            .BeEmpty();
+    }
+
+    [Fact]
+    public async Task Propagates_actor_id_in_handlers_if_able()
+    {
+        await Harness.Start();
+
+        var testPrincipal = new ClaimsPrincipal(
+            new ClaimsIdentity(
+                new Claim[]
+                {
+                    new("sub", "test_id"),
+                    new("role", "user"),
+                    new("role", "admin"),
+                    new("other_claim", "other_claim_value")
+                },
+                TestAuthenticationHandler.SchemeName,
+                "sub",
+                "role"
+            )
+        );
```

Comment:
This can easily be wrapped in a method that takes just pairs of claims. Also, specifying all the claims seems to be unnecessary. It only needs `sub` claim.